tests/system/admin_overlay/test_system_async.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+from google.cloud.bigtable import admin_v2
+from google.cloud.bigtable.data._cross_sync import CrossSync
+from google.cloud.bigtable.data import mutations, read_rows_query
+
+from .conftest import (
+    INSTANCE_PREFIX,
+    BACKUP_PREFIX,
+    ROW_PREFIX,
+    DEFAULT_CLUSTER_LOCATIONS,
+    REPLICATION_CLUSTER_LOCATIONS,
+    TEST_TABLE_NAME,
+    TEST_BACKUP_TABLE_NAME,
+    TEST_COLUMMN_FAMILY_NAME,
+    TEST_COLUMN_NAME,
+    NUM_ROWS,
+    INITIAL_CELL_VALUE,
+    NEW_CELL_VALUE,
+)
+
+from datetime import datetime, timedelta
+
+import asyncio
+import pytest
+import time
+
+
+if CrossSync.is_async:
+    from google.api_core import operation_async as api_core_operation
+else:
+    from google.api_core import operation as api_core_operation
+
+
+__CROSS_SYNC_OUTPUT__ = "tests.system.admin_overlay.test_system_autogen"
+
+
+@CrossSync.drop
+@pytest.fixture(scope="session")
+def event_loop():
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    yield loop
+    loop.stop()
+    loop.close()
+
+
+@CrossSync.convert
+@CrossSync.pytest_fixture(scope="session")
+async def data_client():
+    client = CrossSync.DataClient()
+    yield client
+    await client.close()
+
+
+@CrossSync.convert(
+    replace_symbols={"BigtableTableAdminAsyncClient": "BigtableTableAdminClient"}
+)
+@CrossSync.pytest_fixture(scope="session")
+async def table_admin_client():
+    client = admin_v2.BigtableTableAdminAsyncClient()
+    yield client
+    await client.transport.close()
+
+
+@CrossSync.convert(
+    replace_symbols={"BigtableInstanceAdminAsyncClient": "BigtableInstanceAdminClient"}
+)
+@CrossSync.pytest_fixture(scope="session")
+async def instance_admin_client():
+    client = admin_v2.BigtableInstanceAdminAsyncClient()
+    yield client
+    await client.transport.close()
+
+
+@CrossSync.convert
+@CrossSync.pytest_fixture(scope="session")
+async def instances_to_delete(instance_admin_client):
+    instances = []
+
+    yield instances
+
+    for instance in instances:
+        await instance_admin_client.delete_instance(name=instance.name)
```

Comment:
IIRC I think you might need a try/finally block here (and delete backups), to make sure these are always cleaned up even if there's an exception

```
async def instances_to_delete(instance_admin_client):
    instances = []
    try:
        yield instances
    finally:
        for instance in instances:
            await instance_admin_client.delete_instance(name=instance.name)
```